File: HTTPShortcuts/app/src/main/java/ch/rmy/android/http_shortcuts/plugin/PluginBroadcastReceiver.java
package ch.rmy.android.http_shortcuts.plugin;

import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.support.annotation.NonNull;

import com.twofortyfouram.locale.sdk.client.receiver.AbstractPluginSettingReceiver;

import ch.rmy.android.http_shortcuts.utils.IntentFactory;

public final class PluginBroadcastReceiver extends AbstractPluginSettingReceiver {

    @Override
    protected boolean isBundleValid(@NonNull final Bundle bundle) {
        return PluginBundleManager.isBundleValid(bundle);
    }

    @Override
    protected boolean isAsync() {
        return false;
    }

    @Override
    protected void firePluginSetting(@NonNull final Context context, @NonNull final Bundle bundle) {
        long shortcutId = PluginBundleManager.getShortcutId(bundle);
        Intent intent = IntentFactory.createIntent(context, shortcutId);
        context.startActivity(intent);
    }

}

File: HTTPShortcuts/app/src/main/java/ch/rmy/android/http_shortcuts/plugin/PluginBroadcastReceiver.kt
package ch.rmy.android.http_shortcuts.plugin

import android.content.Context
import android.os.Bundle
import ch.rmy.android.http_shortcuts.activities.ExecuteActivity
import com.twofortyfouram.locale.sdk.client.receiver.AbstractPluginSettingReceiver

class PluginBroadcastReceiver : AbstractPluginSettingReceiver() {

    override fun isBundleValid(bundle: Bundle) = PluginBundleManager.isBundleValid(bundle)

    override fun isAsync() = false

    override fun firePluginSetting(context: Context, bundle: Bundle) {
        val shortcutId = PluginBundleManager.getShortcutId(bundle)
        val variableValues = PluginBundleManager.getVariableValues(bundle)
        val intent = ExecuteActivity.IntentBuilder(context, shortcutId)
                .variableValues(variableValues)
                .build()
        context.startActivity(intent)
    }

}
